tests/testsuite/cargo_add.rs

```diff
@@ -0,0 +1,2114 @@
+pub fn cargo_exe() -> &'static std::path::Path {
+    snapbox::cmd::cargo_bin!("cargo")
+}
+
+pub fn cargo_command() -> snapbox::cmd::Command {
```

Comment:
This seems really unfortunate to duplicate all the code from cargo-test-support.  Can this be shared somehow?

Reply:
Yes, I did this for having a cleaner integration of the code for this PR and have it in the deferred section for cleaning up in a follow up PR.